src/components/SubscriptionsPage.vue

```diff
@@ -36,20 +56,70 @@
         </div>
     </div>
     <br />
+
+    <ModalComponent v-if="showCreateGroupModal" @close="showCreateGroupModal = !showCreateGroupModal">
+        <h2 v-t="'actions.create_group'" />
+        <div class="flex flex-col">
+            <input class="input my-4" type="text" v-model="newGroupName" :placeholder="$t('actions.group_name')" />
+            <button class="ml-auto btn w-max" v-t="'actions.create_group'" @click="createGroup()" />
+        </div>
+    </ModalComponent>
+
+    <ModalComponent v-if="showEditGroupModal" @close="showEditGroupModal = false">
+        <h2>{{ selectedGroup.groupName }}</h2>
+        <div class="flex flex-col mt-3 mb-2">
+            <div v-for="subscription in subscriptions" :key="subscription.name">
+                <div class="flex justify-between">
+                    <span>{{ subscription.name }}</span>
+                    <DefaultValueCheckbox
+                        :default-value="selectedGroup.channels.includes(subscription.url.substr(-11))"
+                        :callback="() => checkedChange(subscription)"
+                    />
+                </div>
+                <hr />
+            </div>
+        </div>
+    </ModalComponent>
 </template>
 
 <script>
+import DefaultValueCheckbox from "./DefaultValueCheckbox.vue";
+import ModalComponent from "./ModalComponent.vue";
+
 export default {
     data() {
         return {
             subscriptions: [],
+            selectedGroup: {
+                groupName: "",
+                channels: [],
+            },
+            channelGroups: [],
+            showCreateGroupModal: false,
+            showEditGroupModal: false,
+            newGroupName: "",
         };
     },
     mounted() {
         this.fetchSubscriptions().then(json => {
             this.subscriptions = json;
             this.subscriptions.forEach(subscription => (subscription.subscribed = true));
         });
+
+        this.channelGroups = this.channelGroups.concat(this.selectedGroup);
+
+        if (!window.db) return;
+        const cursor = this.getChannelGroupsCursor();
+        cursor.onsuccess = e => {
+            const cursor = e.target.result;
+            if (cursor) {
+                const group = cursor.value;
+                this.channelGroups = this.channelGroups.concat({
+                    groupName: group.groupName,
+                    channels: JSON.parse(group.channels),
+                });
```

Comment:
Why not just `this.channelGroups.push`? I noticed that if we have more than 1 channel group, only one seems to show, despite being in the database 🤔

Reply:
There were some `cursor.continue` calls missing which caused that. 
I didn't use push because I was still used to Svelte, which requires you to assign a new value in order to update. (push wouldn't work in Svelte).